shops/views.py

```diff
@@ -33,3 +36,43 @@ def get_queryset(self):
 class ShopDetailView(generic.DetailView):
     model = Shop
     template_name = 'shops/shopdetail.html'
+
+
+def city_form(request):
+    if request.method == 'POST':
+        form = CityForm(request.POST)
+        if form.is_valid():
+            name = form.cleaned_data['name']
+            new_city = City(name=name)
+            new_city.save()
+            return HttpResponseRedirect(reverse('city'))
+    else:
+        form = CityForm()
+
+    return render(request,'shops/cityform.html', {'form': form})
+
+
+def shop_form(request):
+    if request.method == 'POST':
+        form = ShopForm(request.POST)
+        if form.is_valid():
+            city = form.cleaned_data['city']
+            name = form.cleaned_data['name']
+            code = form.cleaned_data['code']
+            address = form.cleaned_data['address']
+            postcode = form.cleaned_data['postcode']
+            year_opened = form.cleaned_data['year_opened']
```

Comment:
You could move all of this into 'new_shop = Shop( .... ) and not have to create a temporary variables. It won't make your code less readable, and possibly even easier to follow